bgpd/rfapi/rfapi_import.c

```diff
@@ -2348,19 +2347,27 @@ static void rfapiWithdrawTimerVPN(struct thread *t)
 	const struct prefix *p;
 	struct rfapi_monitor_vpn *moved;
 	afi_t afi;
+	bool early_exit = false;
 
 	if (bgp == NULL) {
 		vnc_zlog_debug_verbose(
                    "%s: NULL BGP pointer, assume shutdown race condition!!!",
                    __func__);
-		return;
+		early_exit = true;
 	}
-	if (CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS)) {
+	if (bgp && CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS)) {
 		vnc_zlog_debug_verbose(
 			"%s: BGP delete in progress, assume shutdown race condition!!!",
 			__func__);
+		early_exit = true;
+	}
+
+	/* This callback is responsible for the withdraw object's memory */
+	if (early_exit) {
+		XFREE(MTYPE_RFAPI_WITHDRAW, wcb);
```

Comment:
Btw, shouldn't we call `goto done;` instead?

Reply:
my read was that at shutdown the other objects that are being handled in the `done` label are being cleaned up, and that the issue was with the callback-context object  - that that is the only memory that no one else will take responsibility for.